lib/internal/Magento/Framework/Logger/Handler/Base.php

```diff
@@ -34,11 +34,15 @@ class Base extends StreamHandler
      */
     public function __construct(
         DriverInterface $filesystem,
-        $filePath = null
+        $filePath = null,
+        $fileName = null
     ) {
         $this->filesystem = $filesystem;
+        if(null === $fileName){
+            $fileName = $this->fileName;
+        }
         parent::__construct(
-            $filePath ? $filePath . $this->fileName : BP . $this->fileName,
+            $filePath ? $filePath . $fileName : BP . $fileName,
```

Comment:
This would not be needed if you would've changed the object property instead. See the above comment.